apps/web/src/components/settings/AccountSettings.tsx

```diff
@@ -5,33 +5,55 @@ import { Button } from "@/components/shadcn/ui/button";
 import { Label } from "@/components/shadcn/ui/label";
 import { toast } from "sonner";
 import { useState } from "react";
-import { users } from "db/schema";
-import { z } from "zod";
 import { useAction } from "next-safe-action/hook";
 import { modifyAccountSettings } from "@/actions/user-profile-mod";
+import { Checkbox } from "@/components/shadcn/ui/checkbox";
+import c from "config";
+import { Loader2 } from "lucide-react";
 
 interface UserProps {
 	firstName: string;
 	lastName: string;
+	email: string;
+	hackerTag: string;
+	hasSearchableProfile: boolean;
 }
+
 interface AccountSettingsProps {
 	user: UserProps;
 }
 
 export default function AccountSettings({ user }: AccountSettingsProps) {
 	const [newFirstName, setNewFirstName] = useState(user.firstName);
 	const [newLastName, setNewLastName] = useState(user.lastName);
+	//const [newEmail, setNewEmail] = useState(user.email);
+	const [newHackerTag, setNewHackerTag] = useState(user.hackerTag);
+	const [newIsProfileSearchable, setNewIsProfileSearchable] = useState(
+		user.hasSearchableProfile,
+	);
+	const [hackerTagTakenAlert, setHackerTagTakenAlert] = useState(false);
+
+	const [isLoading, setIsLoading] = useState(false);
```

Comment:
nit and non-blocking: useAction should come with loading state as well. However, our action is a bit older here so that may not be the case.